### PR TITLE
Correct check for getenv()/secure_getenv()

### PR DESCRIPTION
**Pull request by ahmedyarub**
I encountered this building for Android. Android doesn't have neither function.

### CODE REVIEW

File: kubernetes/config/kube_config.c [hunk 1]
```diff
@@ -111,19 +111,18 @@ static char *getWorkingConfigFile(const char *configFileNamePassedIn)
         configFileName = strdup(configFileNamePassedIn);
     } else {
 
-#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__) || !defined(HAVE_SECURE_GETENV)
```
Comment (brendandburns) [inline kubernetes/config/kube_config.c:114]: Please leave this `#if` and nest your `#if` inside of it.

File: kubernetes/config/kube_config.c [hunk 2]
```diff
@@ -111,19 +111,18 @@ static char *getWorkingConfigFile(const char *configFileNamePassedIn)
...
-#elif __linux || defined(__EMSCRIPTEN__)
+#if defined(HAVE_SECURE_GETENV)
         kubeconfig_env = secure_getenv(ENV_KUBECONFIG);
+#elif defined(HAVE_GETENV)
+        kubeconfig_env = getenv(ENV_KUBECONFIG);
 #endif
         if (kubeconfig_env) {
             configFileName = strdup(kubeconfig_env);
         } else {
-#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__) || !defined(HAVE_SECURE_GETENV)
```
Comment (brendandburns) [inline kubernetes/config/kube_config.c:122]: same as above.

**Comment by brendandburns** [comment]
Minor comments, and CI is failing.

File: kubernetes/config/kube_config.c (hunk 1 above)
Comment (ahmedyarub) [reply kubernetes/config/kube_config.c:114]: I don't think that that would work. Checking for the operating system is not enough. A much more reliable solution is to check whether the symbol exists in the compilation target. That renders the #if above redundant.

**Comment by ahmedyarub** [comment]
> Minor comments, and CI is failing.

It's weird because on my machine the tests do pass! I'll keep on investigating...

**Comment by ahmedyarub** [comment]
I introduced function/symbol tests last year but I have just found that my implementation is wrong although everything was working! This is the correct implementation for checking getenv, secure_getenv, and strndup.

**Comment by brendandburns** [comment]
This looks good to me. Thanks for the updates.

/lgtm
/approve

**Pull request merged**